src/main/java/com/fasterxml/jackson/core/JsonFactory.java

```diff
@@ -1809,6 +1809,7 @@ protected JsonParser _createParser(InputStream in, IOContext ctxt) throws IOExce
                     e.addSuppressed(e2);
                 }
             }
+            ctxt.close();
```

Comment:
If this is needed, could you move it to its own PR? Seems like a general bug (if it is an issue at all).

Reply:
No this is not a bug in the current implementation because `IOContext` is not `Autocloseable`. I made it so only with this pull request.

Reply:
Ok. That makes rolling this out harder. We will need to track down everywhere that an IOContext is created and ensure that it is closed. This could have a big impact.

Reply:
> Is there any way that we could consider a v2 solution where the IOContexts are registered in a static collection? Meaning we can have a centralised way of closing unclosed IOContexts.
> 
> We can still have the close on the IOContext too and this would include removing the context from the static collection.

This is possible even though will add some (hopefully minimal) performance overhead. However I see 2 more problems with this solution:

1. How can I know if an `IOContext` is a leftover that can be closed or it is still in use? 
2. Who will call the centralized cleanup and when?

Reply:
sure - if the IOContext only needs to be closed when the new BufferRecyclerPool feature is enabled, then maybe that is enough. If we have a bug where the close is not being called when it should then users can go back to the legacy behaviour until we fix the bug.

Reply:
The other alternative would be that the `IOContext` doesn't have a reference to a `BufferRecycler` at all. For each and every operation that needs a `BufferRecycler` it will take it from the pool and put it back into the pool immediately after, when it's done. This will increase the contention on the pool (maybe not a big issue with a well implemented pool), but will have the twofold advantage of not requiring any `close` method and also increasing the reuse of the `BufferRecycler`s themselves.

@pjfanning If you think this is more acceptable (and safe) I could try to implement it with a separated pull request. 
@cowtowncoder @franz1981 Feedback is welcome :)

Reply:
Nope, I'm realizing that what I suggested in my last comment cannot work (or I think so) because doing so the same `BufferRecycler` will be used by different contexts at the same time :( If so please feel free to ignore it.